### PR TITLE
add ssl config

### PR DESCRIPTION
**Pull request by cristi1979**
- the rest of the changes are from Geppetto autoformat

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -109,32 +124,30 @@
 
   file { '/etc/default/pulp_workers':
     ensure  => file,
-    content => template("pulp/${pulp::pulp_workers_template}"),
+    content => template("pulp/${pulp::params::pulp_workers_template}"),
```
Comment (ekohl) [inline manifests/config.pp]: This doesn't seem right.

File: manifests/config.pp [hunk 2]
```diff
@@ -109,32 +124,30 @@
...
   if $pulp::reset_cache {
-    exec {'reset_pulp_cache':
+    exec { 'reset_pulp_cache':
       command => 'rm -rf /var/lib/pulp/packages/*',
       path    => '/sbin:/bin:/usr/bin',
       before  => Exec['migrate_pulp_db'],
-      require => [
-        File['/var/lib/pulp/packages'],
-        ],
+      require => [File['/var/lib/pulp/packages'],],
```
Comment (ekohl) [inline manifests/config.pp]: Might as well change it to ```require => File['/var/lib/pulp/packages'],``` then

**Comment by cristi1979** [comment]
- this configuration replaces pulp.conf.erb
- also adds firewall rules

**Comment by ekohl** [comment]
In general it's uncommon to set firewall rules in a module. It's usually done in profiles and I'd certainly think it needs to be in a separate PR.

File: metadata.json [hunk 3]
```diff
@@ -16,7 +16,6 @@
        "name": "puppetlabs/apache",
        "version_requirement": ">= 1.0.0 < 2.0.0"
     },
-    {
```
Comment (ekohl) [inline metadata.json]: Doesn't this break the json file?

File: manifests/config.pp [hunk 4]
```diff
@@ -22,15 +21,72 @@
     mode    => '0600',
   }
 
-  file {'/etc/httpd/conf.d/pulp.conf':
```
Comment (ekohl) [inline manifests/config.pp]: If you're moving to using ```apache::vhost``` (which I think is a good move), then you should remove the template (```templates/etc/httpd/conf.d/pulp.conf.erb```) as well.

File: manifests/config.pp [hunk 5]
```diff
@@ -22,15 +21,72 @@
...
+    priority            => '05',
+    docroot             => '/srv/pulp',
+    port                => 80,
+    servername          => [$::fqdn],
+    serveraliases       => [$::hostname],
+    additional_includes => '/etc/pulp/vhosts80/*.conf',
+    add_default_charset => 'UTF-8',
+  }
+
+  apache::vhost {'pulp-https':
```
Comment (ekohl) [inline manifests/config.pp]: I can't see you setting the ```WSGISocketPrefix```. Maybe this is done while configuring the module, but it's important to note.

File: manifests/config.pp [hunk 6]
```diff
@@ -22,15 +21,72 @@
...
   }
 
-  file {'/etc/httpd/conf.d/pulp.conf':
+  apache::vhost { 'pulp-http':
+    priority            => '05',
+    docroot             => '/srv/pulp',
+    port                => 80,
+    servername          => [$::fqdn],
+    serveraliases       => [$::hostname],
+    additional_includes => '/etc/pulp/vhosts80/*.conf',
```
Comment (ekohl) [inline manifests/config.pp]: This is only done for Fedora 18, which maybe should be dropped. @ehelms?

File: manifests/config.pp [hunk 7]
```diff
@@ -22,15 +21,72 @@
...
+        'path'     => 'webservices.wsgi',
+        'provider' => 'files',
+      },
+      {
+        'path'     => '/srv/pulp',
+        'provider' => 'directory',
+      }
+      ],
+    options                    => ['SymLinksIfOwnerMatch'],
+    custom_fragment            => '  SSLInsecureRenegotiation on'
```
Comment (ekohl) [inline manifests/config.pp]: The comment (allow older yum clients to connect, see bz 647828) is helpful to understand why it's needed and I'd at least leave it here.

File: manifests/config.pp (hunk 5 above)
Comment (cristi1979) [inline manifests/config.pp]: WSGISocketPrefix is set from apache::mod::wsgi class, so I didn't thought it's necessary to set it explicitly.

**Comment by cristi1979** [comment]
- removed firewall rules
- removed old fact (mongodb_version)
- fix apache definition

File: .fixtures.yml [hunk 8]
```diff
@@ -2,6 +2,7 @@ fixtures:
   repositories:
     stdlib:        "git://github.com/puppetlabs/puppetlabs-stdlib.git"
     foreman:       "git://github.com/theforeman/puppet-foreman.git"
+    firewall:      "git://github.com/puppetlabs/puppetlabs-firewall"
```
Comment (ekohl) [inline .fixtures.yml]: This is no longer needed.

File: lib/facter/facts.rb [hunk 9]
```diff
@@ -1,16 +0,0 @@
-Facter.add(:mongodb_version) do
```
Comment (ehelms) [inline lib/facter/facts.rb]: Is this removal directly related to this change?

File: manifests/config.pp [hunk 10]
```diff
@@ -22,15 +21,72 @@
...
+    aliases                    => [{
+        alias           => '/pulp/static',
+        path            => '/var/lib/pulp/static',
+        options         => ['Indexes'],
+        custom_fragment => 'SSLRequireSSL'
+      }
+      ],
+    options                    => ['SymLinksIfOwnerMatch'],
+    custom_fragment            => '# allow older yum clients to connect, see bz 647828
+  SSLInsecureRenegotiation on'
```
Comment (ehelms) [inline manifests/config.pp]: LInes 69-78 seem off indentation wise.

**Comment by cristi1979** [comment]
I'm not sure how to push some other changes, because the files are already modified by other PRs.

I would like to add 2 new parameters:

     $ssl_enforce_verify = true
     $enable_http = false

$ssl_enforce_verify should allow https access to the repos without requiring authenticating with the certificates (browsers and yum without ssl certificates).

**Comment by cristi1979** [comment]
http access is also required for puppet repos, at least.

File: manifests/config.pp (hunk 1 above)
Comment (ehelms) [inline manifests/config.pp]: Agreed, we should avoid the use of params

File: manifests/config.pp [hunk 11]
```diff
@@ -22,15 +21,72 @@
...
   }
 
-  file {'/etc/httpd/conf.d/pulp.conf':
+  apache::vhost { 'pulp-https':
+    priority                   => '05',
+    docroot                    => '/srv/pulp',
+    port                       => 443,
+    servername                 => [$::fqdn],
+    serveraliases              => [$::hostname],
+    add_default_charset        => 'UTF-8',
```
Comment (ehelms) [inline manifests/config.pp]: I am getting a `Invalid parameter add_default_charset` when testing this.

Comment (cristi1979) [inline manifests/config.pp]: I see it was added in apache 1.2.0. Should I remove it?

File: manifests/config.pp (hunk 1 above)
Comment (cristi1979) [inline manifests/config.pp]: Sorry about that, I will remove params.

File: manifests/config.pp (hunk 11 above)
Comment (ehelms) [inline manifests/config.pp]: If we keep it, we'd need to bump the metadata requirement, however, due to how we use this module currently I'd prefer to stay 1.1.0 compatible.

**Comment by cristi1979** [comment]
rebased on the original config.pp

File: manifests/config.pp [hunk 12]
```diff
@@ -14,12 +14,58 @@
...
+        custom_fragment => 'SSLRequireSSL'
+      }
+    ],
+    options             => ['SymLinksIfOwnerMatch'],
+    custom_fragment     => '# allow older yum clients to connect, see bz 647828
+  SSLInsecureRenegotiation on
+  AddDefaultCharset UTF-8'
   }
 
   file {'/etc/httpd/conf.d/pulp.conf':
```
Comment (ehelms) [inline manifests/config.pp]: Looks like this needs to be removed.

File: manifests/config.pp [hunk 13]
```diff
@@ -14,12 +14,58 @@
     mode   => '0755';
   }
 
-  file {'/etc/pulp/server.conf':
```
Comment (ehelms) [inline manifests/config.pp]: DId you mean to remove this one or the one I pointed out below?

File: manifests/config.pp [hunk 14]
```diff
@@ -14,12 +14,58 @@
...
-    ensure  => file,
-    content => template('pulp/etc/pulp/server.conf.erb'),
-    owner   => 'apache',
-    group   => 'apache',
-    mode    => '0600',
+  apache::vhost { 'pulp-https':
```
Comment (ehelms) [inline manifests/config.pp]: What do you think about putting this in it's own class? `pulp::apache`

**Comment by ehelms** [comment]
@cristi1979 Apologies for the long delay in getting back to these, releases and other things got me backed up. Hopefully you haven't given up on these; however, if you need someone to take them over for you please let us know.

**Comment by cristi1979** [comment]
closing. I will do a new PR
